.cargo/config.toml

```diff
@@ -10,9 +10,24 @@ lto = true
 # https://blog.rust-lang.org/2023/11/16/Rust-1.74.0.html#lint-configuration-through-cargo
 rustflags = [
 	# rustc additional warnings:
-	"--warn",
-	"unused_crate_dependencies",
-	# clippy additional warnings:
-	"--warn",
-	"clippy::wildcard_imports",
+	"-Wunused_crate_dependencies",
+	# Rust 2018 idioms that are not yet warn-by-default:
+	"-Welided_lifetimes_in_paths",
+	"-Wunused_extern_crates",
+	"-Wexplicit_outlives_requirements",
+	# 📎 Lints that are enabled (warn/deny) by default
+	"-Wclippy::all",
+	# Restriction (optional, neutral lints)
+	"-Wclippy::exit",                            # Prefer not `process::exit`ing directly
```

Comment:
A few more suggestions I found while reviewing. We can definetely just add them to the issue, instead of piling more in the same PR. You have already done a lot of useful work here:

- clippy::cargo	(entire category)
- clippy::unimplemented (should we add it in CI next to `todo` and `dbg`?)
- clippy::same_name_method
- clippy::use_self
- clippy::allow_attributes_without_reason
- clippy::self_named_module_files

And less importantly:

- clippy::collection_is_never_read
- clippy::equatable_if_let
- clippy::format_push_string
- clippy::str_to_string
- clippy::useless_let_if_seq
- clippy::wildcard_enum_match_arm

I think these are all the other "allowed" rules that I found useful in this project. But maybe it is too strict?

Reply:
We have a legitimate usage for `unimplemented`: https://github.com/Xanewok/slang/blob/25b624b790a35701cf6b705a855734657566f477/crates/codegen/parser/runtime/src/support/repetition_helper.rs#L17

Also, from the docs:

> The difference between `unimplemented!` and [`todo`] is that while todo! conveys an intent of implementing the functionality later and the message is "not yet implemented", unimplemented! makes no such claims. Its message is "not implemented". Also some IDEs will mark `todo!`s.

As for the rest, I'll leave the remainder for later, as you suggest, if you don't mind :pray: There's quite a bit already here.

However, I would probably not lint against `nursery` lints unless very beneficial for us, since it's inherently unstable by nature and sometimes might trip us or maybe Clippy folks will deem it unnecessary/-ily opinionated.